src/extension/extension.ts

```diff
@@ -271,7 +272,25 @@ export async function activate(context: vs.ExtensionContext, isRestart: boolean
 			};
 		}
 
-		flutterDaemon = new FlutterDaemon(logger, workspaceContext as FlutterWorkspaceContext, flutterCapabilities, runIfNoDevices);
+		if (workspaceContext.config.forceFlutterWorkspace) {
+			let resultFromLocalExtension = null;
+			try {
+				resultFromLocalExtension = await vs.commands.executeCommand<string>("dartlocaldevice.startDaemon", workspaceContext.config.flutterToolsScript?.script, "expose_devices", workspaceContext.config.flutterSdkHome);
```

Comment:
It might also be worth either having a timeout on this, or showing a progress indicator that allows the user to stop waiting for it if it takes too long (since if we get stuck here, it'll prevent the rest of the extension activating).

Reply:
I moved the timeout function to utils; let me know if you intended to keep that test-specific.

Reply:
Ah makes sense. I'll keep that mind and update depending on user feedback.